### DISCUSSION TITLE
Why is the Gradle EvalTask marked as @UntrackedTask?

### DISCUSSION THREAD

**Discussion by odenix** [apple/pkl#335] [General]
The rationale given is that "Output file names are known only after execution". However, the `getOutputFile` and `getMultipleFileOutputDir` task properties seem to provide exactly the information that Gradle needs to track the task.

**Comment by holzensp** [comment]
```
output {
  files {
    ...someFunction(read("env:SOMEVAR"))
  }
}
```
^^ There's no telling what output files you'll have without evaluating the module.

**Reply by StefMa** [reply]
Aren't gradle provider/properties types for exactly this purpose? 🤔

After evaluation is not the same as after execution 🙃

**Reply by odenix** [reply]
> ^^ There's no telling what output files you'll have without evaluating the module.

All output files will be located under `getMultipleFileOutputDir`. I don’t see how this is different from (say) the `JavaCompile` task, where all that’s known upfront is that all output files will be located under `destinationDir`.

**Comment by odenix** [comment]
Sent: https://github.com/apple/pkl/pull/390

**Comment by holzensp** [comment]
Resolved by #390
